hil/ext/switches/ovs.py

```diff
@@ -0,0 +1,261 @@
+            port: switch port in a valid format
+            channel: eg 'vlan/native' or 'vlan/<vlan_id>
+            new_network: vlan_id
+        Returns: If successful returns None else error.
+        """
+        (port,) = filter(lambda p: p.label == port, self.ports)
+        interface = port.label
+
+        if channel == 'vlan/native':
+            if new_network is None:
+                return self._remove_native_vlan(interface)
+            else:
+                return self._set_native_vlan(interface, new_network)
+        else:
+            match = re.match(re.compile(r'vlan/(\d+)'), channel)
+            assert match is not None, "HIL passed an invalid channel to the" \
+                " switch!"
+            vlan_id = match.groups()[0]
+
+            if new_network is None:
+                return self._remove_vlan_from_port(interface, vlan_id)
+            else:
+                assert new_network == vlan_id
+                return self._add_vlan_to_trunk(interface, vlan_id)
+
+# 2. Private Methods:
+
+    def _string_to_list(self, a_string):
+        """Converts a string representation of list to list.
+        Args:
+            a_string: list output recieved as string.
+                e.g. Strings starting with '[' and ending with ']'
+        Returns: object of list type.
+                 Empty list is put as None.
+        """
+        if a_string == '[]':
+            return ast.literal_eval(a_string)
+        else:
+            a_string = a_string.replace("[", "['").replace("]", "']")
+            a_string = a_string.replace(",", "','")
+            a_list = ast.literal_eval(a_string)
+            a_list = [ele.strip() for ele in a_list]
+            return a_list
+
+    def _string_to_dict(self, a_string):
+        """Converts a string representation of dictionary
+        into a dictionary type object.
+
+        Args:
+            a_string: dictionary recieved as type string
+                eg. Strings starting with '{' and ending with '}'
+        Returns: Object of dictionary type.
+        """
+        if a_string == '{}':
+            a_dict = ast.literal_eval(a_string)
+            return a_dict
+        else:
+            a_string = a_string.replace("{", "{'").replace("}", "'}")
+            a_string = a_string.replace(":", "':'").replace(",", "','")
+            a_dict = ast.literal_eval(a_string)
+            a_dict = {k.strip(): v.strip() for k, v in a_dict.iteritems()}
+            return a_dict
+
+    def _interface_info(self, port):
+        """Gets latest configuration of port from switch.
+
+        Args:
+            port: Valid port name
+        Returns: configuration status of the port
+        """
+        shell_cmd = "sudo ovs-vsctl list port {port}".format(port=port)
+        args = shlex.split(shell_cmd)
+        try:
+            output = subprocess.check_output(args)
+        except subprocess.CalledProcessError as e:
+            logger.error(" %s ", e)
+            raise SwitchError
```

Comment:
ditto, again, I am not sure whether it makes sense or not to give a more verbose error message here. Just an idea.

Reply:
next commit fixes it.